test/test_transformers.py

```diff
@@ -1423,7 +1429,7 @@ def test_flash_autocast_fp32_bfloat16(self, device):
                 _ = torch.nn.functional.scaled_dot_product_attention(
                     q, k, v, None, 0.0, False)
 
-    @parametrize("kernel", [SDPBackend.MATH, SDPBackend.FLASH_ATTENTION, SDPBackend.EFFICIENT_ATTENTION])
+    @parametrize("kernel", [SDPBackend.MATH] + PLATFORM_SPECIFIC_SDPA)
```

Comment:
This should be left as is since these are error cases they should error regardless of error, unless the order of SDP_utils is causing different errorrs to be thrown, is that the case?

Reply:
ROCM's situation is a little different with CUDA. on CUDA supporting Flash attention implies supporting memory efficient attention, but on ROCM we decided to support Flash attention first and leave memory efficient attention later. 

This is also why we created `get_platform_specific_sdpa()` function earlier. Specific to this case we cannot test Memory efficient attention on ROCM for now because it only triggers errors.

I'll check and fix the coverage of tests on CUDA platform in a later patch.